A collection of resources and papers for absolute beginners
===========================================================



The goal of this list is to provide a bunch of resources for people working in
the wet-lab, but interested in learning some basic scripting, data cleaning and
analysis to be more efficient. Suggestions and comments are welcome.

Papers with basic info
----------------------

* So you want to be a computational biologist?
  [10.1038/nbt.2740](http://dx.doi.org/10.1038/nbt.2740)
* A quick guide for developing effective bioinformatics programming skills
  [10.1371/journal.pcbi.1000589](http://dx.doi.org/10.1371/journal.pcbi.1000589)
* A quick guide to organizing computational biology projects
  [10.1371/journal.pcbi.1000424](http://dx.doi.org/10.1371/journal.pcbi.1000424)

Work a bit more efficiently and reproducibly
--------------------------------------------

It might seem too much to demand reproducibility, version control and similar
stuff from absolute beginners, but better start early. If you keep a lab
notebook, you should keep something similar when writing ugly hacks. Probably
the information is quite redundant in the following papers, but just to make
sure.

* A statistical definition for reproducibility and replicability
  [10.1101/066803](http://dx.doi.org/10.1101/066803)
* Five selfish reasons to work reproducibly
  [10.1186/s13059-015-0850-7](http://dx.doi.org/10.1186/s13059-015-0850-7)
* A Quick Introduction to Version Control with Git and GitHub
  [10.1371/journal.pcbi.1004668](http://dx.doi.org/10.1371/journal.pcbi.1004668)
* Ten Simple Rules for Taking Advantage of git and GitHub
  [10.1101/048744](http://dx.doi.org/10.1101/048744)
* Tidy Data [10.18637/jss.v059.i10](http://dx.doi.org/10.18637/jss.v059.i10)
* Best Practices for Scientific Computing
  [10.1371/journal.pbio.1001745](http://dx.doi.org/10.1371/journal.pbio.1001745)
* Good Enough Practices in Scientific Computing
  [arXiv:1609.00037](http://arxiv.org/abs/1609.00037)
* Ten Simple Rules for Reproducible Computational Research
  [10.1371/journal.pcbi.1003285](http://dx.doi.org/10.1371/journal.pcbi.1003285)

Perspective
-----------

* Biological data sciences in genome research
  [10.1101/gr.191684.115](http://dx.doi.org/10.1101/gr.191684.115)

Primer collections
------------------

* An Introduction to Programming for Bioscientists: A Python-Based Primer
  [10.1371/journal.pcbi.1004867](http://dx.doi.org/10.1371/journal.pcbi.1004867)

Collections of papers with a very short introduction on a diverse set of
computational biology topics.

* Nature Biotechnology and PLoS Computational Biology Primers collected
  [here](http://liacs.leidenuniv.nl/~hoogeboomhj/mcb/nature_primer.html)

Collection of papers in Nature Methods on data visualization.

* Data visualization: A view of every Points of View column
  [here](http://blogs.nature.com/methagora/2013/07/data-visualization-points-of-view.html)

Collection of papers in Nature Methods on basic statistical concepts.

* Points of Significance: Statistics for Biologists
  [here](http://www.nature.com/collections/qghhqm/pointsofsignificance)

Stuff to install before starting the courses and books
------------------------------------------------------

* Install some kind of Linux distribution and play around. Cry a lot. Maybe
  [Ubuntu](http://www.ubuntu.com/).
* Install [R](https://www.r-project.org/) and
  [RStudio](https://www.rstudio.com/).
* Install [Anaconda](https://www.continuum.io/downloads), a Python distribution.
* Check [bioconda](https://bioconda.github.io) and install some bioinformatics
  packages.
* Check [Bioconductor](https://bioconductor.org/install) and install some R packages.

Linux command line
------------------

This might be useful right after installing Ubuntu and before starting to learn
anything else.

* The Command Line Crash [Course](http://cli.learncodethehardway.org/book)

Online courses
--------------

* Data Science Specialization at
  [Coursera](https://www.coursera.org/specializations/jhu-data-science). This is
  a general course on cleaning up, exploring, plotting and analysing data.
* Genomic Data Science Specialization at
  [Coursera](https://www.coursera.org/specializations/genomic-data-science).
  Tools and methods to analyse sequencing data, from Galaxy, to the linux
  command line and python or R scripts.
* Rosalind, a [platform for learning bioinformatics](http://rosalind.info/problems/locations)
  through problem solving
* Software Carpentry [Lessons](http://software-carpentry.org/lessons) with lots
  of basic info on the Linux shell, programming, git, automation,
  reproducibility. The two most important at the start might be the one on [unix
  shell](http://swcarpentry.github.io/shell-novice/) and on [version control
  with git](http://swcarpentry.github.io/git-novice/)

Books and university courses
----------------------------

* [Learn Python the Hard Way](http://learnpythonthehardway.org)
* [Python for Biologists](http://pythonforbiologists.com)
* [R for Data Science](http://r4ds.had.co.nz)
* [Bioinformatics Data Skills](http://shop.oreilly.com/product/0636920030157.do)

More detailed courses focusing on data science, bioinformatics and genomics

* [Introduction to Data Science](https://rafalab.github.io/dsbook/)
* [Introduction to Computational Biology](https://biodatascience.github.io/compbio/)
* [PH525x series - Biomedical Data Science](http://genomicsclass.github.io/book/)
* [Modern Statistics for Modern Biology](https://www.huber.embl.de/msmb/)
